lib/spack/spack/build_systems/lua.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+from typing import List, Text
+
+from llnl.util.filesystem import find, join_path
+
+from spack.package import PackageBase
+from spack.pkgkit import *
+from spack.util.executable import Executable
+
+
+class LuaPackage(PackageBase):
+    """Specialized class for lua packages
+
+    Note that there is also a LuaRocksPackage for the common case where luarocks is the
+    build system"""
+
+    phases = ['preprocess', 'install']
+    #: This attribute is used in UI queries that need to know the build
+    #: system base class
+    build_system_class = 'LuaPackage'
+
+    depends_on('lua-lang')
+    extends('lua', when='^lua')
+    extends('lua-luajit', when='^lua-luajit')
+    depends_on('lua-luajit+lualinks', when='^lua-luajit')
+    extends('lua-luajit-openresty', when='^lua-luajit-openresty')
+    depends_on('lua-luajit-openresty+lualinks', when='^lua-luajit-openresty')
+
+    rocks_build_flags = []  # type: List[Text]
+
+    def preprocess(self, spec, prefix):
+        """Extend this to preprocess source before building with luarocks"""
+        pass
+
+    @property
+    def lua(self):
+        return Executable(join_path(self.spec['lua-lang'].prefix.bin, 'luarocks'))
```

Comment:
Is this a typo:
```suggestion
        return Executable(join_path(self.spec['lua-lang'].prefix.bin, 'lua'))
```
?

Reply:
Yes, that's a typo, thanks!